api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/pooling/GlobalAvgPool1D.kt

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2020 JetBrains s.r.o. and Kotlin Deep Learning project contributors. All Rights Reserved.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlinx.dl.api.core.layer.pooling
+
+import org.jetbrains.kotlinx.dl.api.core.KGraph
+import org.jetbrains.kotlinx.dl.api.core.layer.Layer
+import org.jetbrains.kotlinx.dl.api.core.util.TF
+import org.tensorflow.Operand
+import org.tensorflow.Shape
+import org.tensorflow.op.Ops
+
+/**
+ * Global average pooling operation for temporal data.
+ *
+ * NOTE: Works with tensors which must have rank 3 (batch, steps, features).
+ *
+ * Input shape: 3D tensor with shape `(batch_size, steps, features)`.
+ *
+ * Output shape: 2D tensor with shape `(batch_size, features)`.
+ *
+ * @property [name] Custom layer name.
+ * @constructor Creates [GlobalAvgPool1D] object.
+ *
+ */
+
+class GlobalAvgPool1D(
+        name: String = ""
+) : Layer(name) {
+    override fun build(tf: Ops, kGraph: KGraph, inputShape: Shape) { }
+
+    override fun computeOutputShape(inputShape: Shape): Shape {
+        return Shape.make(inputShape.size(0), inputShape.size(2))
+    }
+
+    override fun forward(tf: Ops, input: Operand<Float>, isTraining: Operand<Boolean>, numberOfLosses: Operand<Float>?): Operand<Float> {
+        var stepAxis = 1                //if dataFormat == "channels_last", it will be 2 if "channels_first"
```

Comment:
please mark it with the TODO: for the future (when both formats will be supported)